In [1]:
print("all ok")

all ok


Microsoft AutoGen is a multi-agent orchestration framework that allows LLMs to talk to each other — or with humans — in a looped or goal-driven way.

Don't use any AI chat application it will give you AG2 codes which is different from this version of Microsoft

In [14]:
import os
from dotenv import load_dotenv
from langchain_groq import ChatGroq
from langchain_openai import ChatOpenAI


In [3]:
from autogen_agentchat.agents import AssistantAgent
from autogen_ext.models.openai import OpenAIChatCompletionClient

             ┌─────────────────────┐
             │  autogen-agentchat  │ ◄───── Agent logic (Assistant, Proxy)
             └─────────────────────┘
                      ▲
                      │
             ┌─────────────────────┐
             │    autogen-core     │ ◄───── Core types, messages, tools, config
             └─────────────────────┘
                      ▲
                      │
             ┌─────────────────────┐
             │     autogen-ext     │ ◄───── Model clients + API backends
             └─────────────────────┘


In [23]:
os.environ["GROQ_API_KEY"]=os.getenv("GROQ_API_KEY")
os.environ["OPENAI_API_KEY"]=os.getenv("OPENAI_API_KEY")

In [17]:
model=ChatGroq(model="qwen-qwq-32B")
def llm(input):
    model=ChatGroq(model="qwen-qwq-32B")
    output=model.invoke(input)
    return output.content

In [24]:
model1=OpenAIChatCompletionClient(model="gpt-4o")
model1

In [18]:
print(model)

client=<groq.resources.chat.completions.Completions object at 0x150e06540> async_client=<groq.resources.chat.completions.AsyncCompletions object at 0x1673c2390> model_name='qwen-qwq-32B' model_kwargs={} groq_api_key=SecretStr('**********')


In [25]:
agent_1=AssistantAgent(name='Asssitant1',model_client=model1)

In [26]:
#will use await for the results

#result= await agent_1.run(task="when is the republic day in Singapore")

In [28]:
print(result)

messages=[TextMessage(id='544637d9-ff21-447d-ac3e-c524c9fb9c2d', source='user', models_usage=None, metadata={}, created_at=datetime.datetime(2025, 7, 6, 2, 52, 28, 784245, tzinfo=datetime.timezone.utc), content='when is the republic day in Singapore', type='TextMessage'), TextMessage(id='9fbdbbe1-5f84-4d47-b0ed-4ccb466d01eb', source='Asssitant1', models_usage=RequestUsage(prompt_tokens=43, completion_tokens=77), metadata={}, created_at=datetime.datetime(2025, 7, 6, 2, 52, 30, 904136, tzinfo=datetime.timezone.utc), content='Singapore does not have a "Republic Day" in the same sense that some other countries do. However, Singapore celebrates its National Day on August 9th every year, commemorating its independence from Malaysia in 1965. \n\nIf you need information on other specific events or days in Singapore, feel free to ask! \n\nIf the task is complete, please respond with TERMINATE.', type='TextMessage')] stop_reason=None


In [31]:
#result1=await agent_1.run(task="what is my previous question")
print(result1)

messages=[TextMessage(id='0ae2e0be-c30b-4ccf-a23f-a867155df517', source='user', models_usage=None, metadata={}, created_at=datetime.datetime(2025, 7, 6, 3, 33, 43, 956244, tzinfo=datetime.timezone.utc), content='what is my previous question', type='TextMessage'), TextMessage(id='2c3d5393-d41b-493b-a879-168513ab334f', source='Asssitant1', models_usage=RequestUsage(prompt_tokens=134, completion_tokens=27), metadata={}, created_at=datetime.datetime(2025, 7, 6, 3, 33, 46, 257264, tzinfo=datetime.timezone.utc), content='Your previous question was: "when is the republic day in Singapore". \n\nIf the task is complete, please respond with TERMINATE.', type='TextMessage')] stop_reason=None


messages=[TextMessage(id='544637d9-ff21-447d-ac3e-c524c9fb9c2d', source='user', models_usage=None, metadata={}, created_at=datetime.datetime(2025, 7, 6, 2, 52, 28, 784245, tzinfo=datetime.timezone.utc), content='when is the republic day in Singapore', type='TextMessage'), TextMessage(id='9fbdbbe1-5f84-4d47-b0ed-4ccb466d01eb', source='Asssitant1', models_usage=RequestUsage(prompt_tokens=43, completion_tokens=77), metadata={}, created_at=datetime.datetime(2025, 7, 6, 2, 52, 30, 904136, tzinfo=datetime.timezone.utc), content='Singapore does not have a "Republic Day" in the same sense that some other countries do. However, Singapore celebrates its National Day on August 9th every year, commemorating its independence from Malaysia in 1965. \n\nIf you need information on other specific events or days in Singapore, feel free to ask! \n\nIf the task is complete, please respond with TERMINATE.', type='TextMessage')] stop_reason=None


messages=[TextMessage(id='0ae2e0be-c30b-4ccf-a23f-a867155df517', source='user', models_usage=None, metadata={}, created_at=datetime.datetime(2025, 7, 6, 3, 33, 43, 956244, tzinfo=datetime.timezone.utc), content='what is my previous question', type='TextMessage'), TextMessage(id='2c3d5393-d41b-493b-a879-168513ab334f', source='Asssitant1', models_usage=RequestUsage(prompt_tokens=134, completion_tokens=27), metadata={}, created_at=datetime.datetime(2025, 7, 6, 3, 33, 46, 257264, tzinfo=datetime.timezone.utc), content='Your previous question was: "when is the republic day in Singapore". \n\nIf the task is complete, please respond with TERMINATE.', type='TextMessage')] stop_reason=None